app/src/app/utils/helpers.ts

```diff
@@ -71,26 +77,65 @@ export const boxAroundPoint = (
  * @returns MapGeoJSONFeature[] | undefined - An array of map features or undefined
  */
 export const getFeaturesInBbox = (
-  map: Map | null,
+  map: MaplibreMap | null,
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   brushSize: number,
-  layers: string[] = [BLOCK_HOVER_LAYER_ID]
+  _layers: string[] = [BLOCK_HOVER_LAYER_ID],
+  filterLocked: boolean = true
 ): MapGeoJSONFeature[] | undefined => {
   const bbox = boxAroundPoint(e, brushSize);
+  const {captiveIds, lockedFeatures, mapDocument, checkParentsToHeal, shatterIds, isPainting} =
+    useMapStore.getState();
+
+  const layers = _layers?.length
+    ? _layers
+    : captiveIds.size
+      ? [BLOCK_HOVER_LAYER_ID, BLOCK_HOVER_LAYER_ID_CHILD]
+      : [BLOCK_HOVER_LAYER_ID];
+
+  let features = map?.queryRenderedFeatures(bbox, {layers}) || [];
+  // If captiveIds exist (eg. block view mode) only select those IDs
+  if (captiveIds.size) {
+    features = features.filter(f => captiveIds.has(f.id?.toString() || ''));
+  }
+  // if filtering locked feature, remove those
+  if (filterLocked && lockedFeatures.size) {
+    features = features.filter(f => !lockedFeatures.has(f.id?.toString() || ''));
+  }
 
-  return map?.queryRenderedFeatures(bbox, {layers});
+  // if there is a child layer and parents have been shattered
+  // check if any of the selected IDs are parents
+  if (mapDocument?.child_layer && shatterIds.parents.size) {
```

Comment:
H: Not sure where to put this comment but I managed to get to a state on the test app with a call stack exceeded error and overlapping geoms.

<img width="1440" alt="Screenshot 2024-11-07 at 10 39 17 PM" src="https://github.com/user-attachments/assets/7cd69428-1046-4ab1-803e-1afeafc016e6">

Reply:
Investigating now -- thank you for flagging this!

Reply:
I wasn't able to replicate this exactly but was getting some strange rendering behavior on blur/refocus of the window/tab. I added some safety around that (a new app loading state 'blurred') and cleaned up some rendering subscriptions